### PR TITLE
Algo develop

### PR DESCRIPTION
**Pull request by Suxiaocai**
1. Add knn_cosine_ss, knn_cosine_all, knn_cosine_cv
2. Updated install.sh
3. Divided examples into 4 subfolders.

### CODE REVIEW

File: algorithms/examples/Similarity/knn_cosine_all_file.gsql [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+      CREATE QUERY knn_cosine_all_file (INT topK, FILE f) FOR GRAPH movie {
+        SumAccum<STRING> @predicted_label;
+        
+        Source = {Person.*};        
+        Source = SELECT s
+                 FROM Source:s 
+                 WHERE s.known_label == ""
+                 ACCUM s.@predicted_label = knn_cosine_ss(s, topK)
```
[the remark this answers is not in the document]
Comment (Suxiaocai) [reply algorithms/examples/Similarity/knn_cosine_all_file.gsql]: @xinyu Why? I need to calculate a @predicted_label for every vertex, and then print to file in POST-ACCUM

File: algorithms/examples/Similarity/knn_cosine_all_attr.gsql [hunk 2]
```diff
@@ -0,0 +1,58 @@
+CREATE QUERY knn_cosine_ss_attr (VERTEX source, INT topK) FOR GRAPH movie RETURNS (STRING) {
```
[the remark this answers is not in the document]
Comment (Suxiaocai) [reply algorithms/examples/Similarity/knn_cosine_all_attr.gsql]: @xchang2014 I also thought about that. My consideration is, if the user revise the subquery of `knn_cosine_all_attr`, then he install `knn_cosine_all`, the subquery would be overwritten. Also, the install.sh is written in this way, so it's hard to change from there.

**Comment by Suxiaocai** [comment]
@xchang2014 I've updated the knn_cosine_all, knn_cosine_cv templates (knn_cosine_ss is already merged to master). Could you review the .gtmp files? After that I can generate the .gsql files, and merge this branch to "reviewed" branch.

File: algorithms/examples/Similarity/knn_cosine_all_file.gsql (hunk 1 above)
[the remark this answers is not in the document]
Comment (xchang2014) [reply algorithms/examples/Similarity/knn_cosine_all_file.gsql]: I meant to put everything under POST-ACCUM

**Pull request merged**